File: android/app/src/main/kotlin/com/example/finwell/DatabaseHelper.kt
package com.example.finwell

import android.database.sqlite.SQLiteDatabase
import android.os.Build
import android.util.Log
import androidx.annotation.RequiresApi
import java.time.LocalDate
import java.time.format.DateTimeFormatter

class DatabaseHelper {



    val LocalDate.day: Int
        @RequiresApi(Build.VERSION_CODES.O)
        get() = this.dayOfMonth

    val LocalDate.monthName: String
        @RequiresApi(Build.VERSION_CODES.O)
        get() = this.format(DateTimeFormatter.ofPattern("MMMM"))

    val LocalDate.year: String
        @RequiresApi(Build.VERSION_CODES.O)
        get() = this.year.toString()
    companion object{
        var instance:DatabaseHelper = DatabaseHelper()
    }
    var db: SQLiteDatabase? = null
    var dbPath = "/data/user/0/com.example.finwell/app_flutter/finwell.db"

    val tableName = "transactions"
    val columnAmount = "TransactionAmount"
    val columnCategory = "TransactionCategory"
    val columnDate = "TransactionDate"
    val columnTransactionId = "TransactionId"
    val columnTransactionName = "TransactionName"
    val columnTransactionType = "TransactionType"
    val columnTransactionStatus = "TransactionStatus"
    var columnTransactionMonth = "TransactionMonth"
    var columnTrasactionYear = "TransactionYear"



    private fun openDataBase(){

        if (db == null){
            db = SQLiteDatabase.openDatabase(dbPath,null,SQLiteDatabase.OPEN_READWRITE)
            Log.e("db","$db")
        }
    }

    @RequiresApi(Build.VERSION_CODES.O)
    fun insertTransaction(amount:String, isDebited:Boolean){
        openDataBase()
        val defaultCategory = "Housing"
        val defaultTransactionName = "Jon Doe Transaction"
        val millisecondsSinceEpoch: Long = System.currentTimeMillis()
        val transactionId = millisecondsSinceEpoch.toString()
        var isExpense = "expense";
        val status = 0
        val today = LocalDate.now()

        isExpense = if(isDebited){
            "expense"
        }
        else{
            "income"
        }
try {
    val sql = "INSERT into $tableName ($columnAmount,$columnCategory,$columnDate, $columnTransactionId, $columnTransactionName, $columnTransactionType, $columnTransactionStatus,$columnTransactionMonth,$columnTrasactionYear) VALUES(\"$amount\", \"$defaultCategory\", \"$millisecondsSinceEpoch\",\"$transactionId\",\"$defaultTransactionName\",\"$isExpense\",\"$status\",\"${today.monthName.lowercase()}\",\"${today.year}\\)"
    db!!.execSQL(sql)
}
catch (e:Exception){
    Log.e("insert Exception",e.toString())
}

    }




}

File: lib/core/database/database_helper.dart
import 'dart:io';

import 'package:finwell/feature/transaction/domain/entities/transaction_model.dart';
import 'package:path/path.dart';
import 'package:path_provider/path_provider.dart';
import 'package:sqflite/sqflite.dart';

class DatabaseHelper {
  DatabaseHelper._();

  static final DatabaseHelper _instance = DatabaseHelper._();

  Database? db;
  String dbName = "finwell.db";
  String tableName = "transactions";
  String columnAmount = "TransactionAmount";
  String columnCategory = "TransactionCategory";
  String columnDate = "TransactionDate";
  String columnTransactionId = "TransactionId";
  String columnTransactionName = "TransactionName";
  String columnTransactionType = "TransactionType";
  String columnTransactionStatus = "TransactionStatus";
  String columnTransactionMonth = "TransactionMonth";
  String columnTrasactionYear = "TransactionYear";

  factory DatabaseHelper() {
    return _instance;
  }

  Future<Database> getDb() async {
    db ??= await openDb();
    return db!;
  }

  Future<Database> openDb() async {
    try {
      Directory parentDirectory = await getApplicationDocumentsDirectory();
      String dbPath = join(parentDirectory.path, dbName);

      print("dbPath -> ${dbPath}");

      return openDatabase(dbPath, onCreate: (db, version) {
        db.execute('''
          CREATE TABLE ${tableName} (
    id INTEGER PRIMARY KEY,
    ${columnAmount} TEXT NOT NULL,
    ${columnCategory} TEXT NOT NULL,
    ${columnDate} INTEGER NOT NULL,
    ${columnTransactionId} TEXT,
    ${columnTransactionName} TEXT NOT NULL,
   ${columnTransactionType} TEXT NOT NULL,
   ${columnTransactionMonth} TEXT NOT NULL,
   ${columnTrasactionYear} TEXT NOT NULL,
   ${columnTransactionStatus} BOOLEAN NOT NULL
          );
          ''');
      }, version: 1);
    } catch (e) {
      throw e;
    }
  }

  insertTransaction(TransactionModel? transactiondata) async {
    Database db = await getDb();
    int result = await db.insert(tableName, {
      columnAmount: transactiondata?.transactionAmount.toString(),
      columnTransactionId: transactiondata?.transactionId,
      columnCategory: transactiondata?.transactionCategory,
      columnDate: transactiondata?.transactionDate.millisecondsSinceEpoch,
      columnTransactionName: transactiondata?.transactionName,
      columnTransactionType: transactiondata?.transactionType,
      columnTransactionStatus: 1,
      columnTransactionMonth: transactiondata?.transactionMonth?.toLowerCase(),
      columnTrasactionYear: transactiondata?.transactionYear
    });
    print(result);
  }
}
